modules/tinymce/src/themes/silver/main/ts/ui/dialog/IFrame.ts

```diff
@@ -25,7 +25,35 @@ const getDynamicSource = (initialData: Optional<string>): IFrameSourcing => {
       // TINY-3769: We need to use srcdoc here, instead of src with a data URI, otherwise browsers won't retain the Origin.
       // See https://bugs.chromium.org/p/chromium/issues/detail?id=58999#c11
       if (cachedValue.get() !== html) {
-        Attribute.set(frameComponent.element, 'srcdoc', html);
+        const iframeElement = frameComponent.element as SugarElement<HTMLIFrameElement>;
+        const setSrcdocValue = () => Attribute.set(iframeElement, 'srcdoc', html);
+
+        if (stream) {
+          const iframe = iframeElement.dom;
+          const iframeDoc = Optional.from(iframe.contentDocument);
```

Comment:
None of the code below will operate if the frame hasn't loaded yet, so I suggest resolving these early as part of the opening conditions rather than constant `map` or `fold` when you make use of them

Reply:
Good suggestion 👍, this improvement also made my code more concise and eliminated the need for `lift2`.

Reply:
It's a hard-learned lesson. On textbox.io the original creators made the editor iframe `window` and `document` references optional, similar to what you did here. Imagine if every time you interacted with editor content you had to unwrap an optional. It was the biggest pain point of the entire codebase 🤣